packages/react-components/src/hooks/useECharts/useEChartOptions.ts

```diff
@@ -15,7 +15,18 @@ export const useEChartOptions = (
   option: EChartsOption,
   settings?: SetOptionOpts
 ) => {
+  const optionRef = useRef(option);
+  const settingsRef = useRef(settings);
+
+  useEffect(() => {
+    optionRef.current = option;
+    settingsRef.current = settings;
+  }, [option, settings]);
+
+  const optionString = JSON.stringify(option);
+  const settingsString = JSON.stringify(settings);
+
   useEffect(() => {
-    chartRef.current?.setOption(option, settings);
-  }, [chartRef, option, settings]);
+    chartRef.current?.setOption(optionRef.current, settingsRef.current);
+  }, [chartRef, optionString, settingsString]);
```

Comment:
Why do we need to use refs here instead of calling setOption with the option and settings directly? It's not immediately clear to me.

Reply:
gah! clearly over engineered it, dont think we need any Refs..